procurement_analytic/views/procurement_analytic.xml

```diff
@@ -10,7 +10,8 @@
     <field name="inherit_id" ref="procurement.procurement_form_view"/>
     <field name="arch" type="xml">
         <field name="group_id" position="after">
-            <field name="account_analytic_id" groups="purchase.group_analytic_accounting"/>
+            <field name="account_analytic_id"
+                   groups="sale.group_analytic_accounting,purchase.group_analytic_accounting"/>
```

Comment:
But sale is not a dependency of this module, isn't it? You should make the change in the sale_... module.